dpgen/generator/styles/gromacs.py

```diff
@@ -0,0 +1,125 @@
+try:
+    from gromacs.fileformats.mdp import MDP
+except ImportError:
+    # JZ: Do not excuse users using other eigiens
+    #dlog.info("GromacsWrapper>=0.8.0 is needed for DP-GEN + Gromacs.")
+    pass
+
+
+@ModelDeviEngien.register("gromacs")
+class GromacsEngien(ModelDeviEngien):
+    def make_input(self, iter_index:int, sys_index:int, directory:Iterator[str], conf_name: str, models: List[str]):
+        # JZ: doesn't support dpdata here??
+        _make_model_devi_native_gromacs(iter_index, directory, self.jdata, self.mdata, conf_name, models)
+    
+    def get_running_parameters(self, work_path: str) -> Tuple[str, List[str]]:
+        with open (os.path.join(work_path, 'cur_job.json'), 'r') as fp:
+            cur_job = json.load (fp)
+        gromacs_settings = self.jdata.get("gromacs_settings", {})
+        mdp_filename = gromacs_settings.get("mdp_filename", "md.mdp")
+        topol_filename = gromacs_settings.get("topol_filename", "processed.top")
+        conf_filename = gromacs_settings.get("conf_filename", "conf.gro")
+        index_filename = gromacs_settings.get("index_filename", "index.raw")
+        deffnm = gromacs_settings.get("deffnm", "deepmd")
+        maxwarn = gromacs_settings.get("maxwarn", 1)
+        nsteps = cur_job["nsteps"]
+        lmp_exec = self.mdata['lmp_command']
+        # Angus: lmp_exec name should be changed to model_devi_exec.
+        # We should also change make_dispatcher
+        # For now, I will use this name for gromacs command
+
+        command = "%s grompp -f %s -p %s -c %s -o %s -maxwarn %d" % (lmp_exec, mdp_filename, topol_filename, conf_filename, deffnm, maxwarn)
+        command += "&& %s mdrun -deffnm %s -nsteps %d" %(lmp_exec, deffnm, nsteps) 
+        
+        forward_files = [mdp_filename, topol_filename, conf_filename, index_filename,  "input.json" ]
+        backward_files = ["%s.tpr" % deffnm, "%s.log" %deffnm , 'model_devi.out', 'model_devi.log']
+        return command, forward_files, backward_files, []
+
+    def extract_trajectory(self, directory) -> 'Trajectory':
+        return GromacsTrajectory(self, directory)
+
+class GromacsTrajectory(Trajectory):
+    def get_model_deviations(self) -> np.ndarray:
+        all_conf = np.loadtxt(os.path.join(self.directory, 'model_devi.out'))
+        self.time = all_conf[:, 0]
+        return all_conf[:, 4]
+
+    def get_frame(self, idx: Tuple[int]) -> "Frame":
+        return GromacsFrame(self, idx)
+
+
+class GromacsFrame(Frame):
+    def read_frame(self) -> dpdata.System:
+        fidx = self.idx[0]
+        time = self.trajectory.time[fidx]
+        conf_name = os.path.join(self.trajectory.directory, 'traj', '%d.gromacstrj' % time)
+        type_map = self.trajectory.engien.jdata['type_map']
+        system = dpdata.System(conf_name, type_map=type_map, fmt='gromacs/gro')
+        return system
+
+def _make_model_devi_native_gromacs(iter_index, task_paths, jdata, mdata, cc, models):
```

Comment:
Maybe this could be a function of class GromacsEngien instead of an individual function?

Reply:
I can update the `GromacsEngien` in this PR. But the work for `LAMMPSEngien` is too heavy. I hope someone could help me, maybe in a separate PR.